packages/kit/src/utils/routing.js

```diff
@@ -1,36 +1,72 @@
-/** @param {string} key */
-export function parse_route_id(key) {
+const param_pattern = /^(\.\.\.)?(\w+)(?:=(\w+))?$/;
+
+/** @param {string} id */
+export function parse_route_id(id) {
 	/** @type {string[]} */
 	const names = [];
 
 	/** @type {string[]} */
 	const types = [];
 
+	// `/foo` should get an optional trailing slash, `/foo.json` should not
```

Comment:
I didn't really look at the changes here. any chance this might fix https://github.com/sveltejs/kit/issues/4294 ?

Reply:
I don't _think_ so. Will need to take a closer look at that issue

Reply:
It does not. Fixing that issue will involve some fundamental changes to how `trailingSlash` is handled